src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -229,8 +243,13 @@ AND NOT EXISTS (select prs from p.results prs where prs.assessor.id = p.student.
     @Query("select distinct p from StudentParticipation p where p.student.id = :#{#studentId} and p.exercise in :#{#exercises}")
     List<StudentParticipation> findByStudentIdAndIndividualExercises(@Param("studentId") Long studentId, @Param("exercises") List<Exercise> exercises);
 
-    @Query("select distinct p from StudentParticipation p left join fetch p.submissions s left join fetch s.results r where p.student.id = :#{#studentId} and p.exercise in :#{#exercises}")
-    List<StudentParticipation> findByStudentIdAndIndividualExercisesWithEagerSubmissionsResult(@Param("studentId") Long studentId, @Param("exercises") List<Exercise> exercises);
+    @Query("select distinct p from StudentParticipation p left join fetch p.submissions s left join fetch s.results r where (p.testRun = false or p.testRun is null) and p.student.id = :#{#studentId} and p.exercise in :#{#exercises}")
```

Comment:
```suggestion
    @Query("select distinct p from StudentParticipation p left join fetch p.submissions s left join fetch s.results r where p.testRun = false and p.student.id = :#{#studentId} and p.exercise in :#{#exercises}")
```

Reply:
fixed in 139c8fb547deadc2b10c0e497d91389dcb5f736c